5/writing-tests.md

```diff
@@ -0,0 +1,118 @@
+Writing Tests
+===
+
+Now that we have our test mock set up, we're ready to write some unit tests for our runtime.
+
+Let's start with a simple test that passes by default. Make sure to add a `#[test]` compiler flag followed by your test set up function. Let's use an `assert!(true)` inside the test for now to make it pass by default. 
+
+Don't forget that the test must execute within the context of `TestExternalities`!
+
+```rust
+#[test]
+fn it_works() {
+	with_externalities(&mut build_ext(), || {
+		assert!(true);
+	})
+}
+```
+
+The command for executing this test is: 
+`cargo test substratekitties.rs` or if want to test your entire package, simply run `cargo test -p YOUR_PACKAGE_NAME`.
+
+The test should pass with the following console output: 
+
+```zsh
+running 1 tests
+test template::tests::it_works ... ok
+
+test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
+```
+
+## Test "Create Kitty" Works
+
+We've done the heavy lifting in setting up the test mock. From this point forward, it is quite easy to write the tests to interact with your modules.
+
+So far, your Kitties module provides critical functionalities like the ability for anyone to create a kitty.
+
+Let's write a test to check that `create_kitty_should_work`: 
+
+```rust
+#[test]
+fn create_kitty_should_work() {
+	with_externalities(&mut build_ext(), || {
+		// create a kitty with account #10.
+		assert_ok!(Kitties::create_kitty(Origin::signed(10)));
+
+		// check that there is now 1 kitty in storage
+		assert_eq!(Kitties::all_kitties_count(), 1);
+
+		// check that account #10 owns 1 kitty
+		assert_eq!(Kitties::owned_kitty_count(10), 1);
+		
+		// check that some random account #5 does not own a kitty
+		assert_eq!(Kitties::owned_kitty_count(5), 0);
+
+		// check that this kitty is specifically owned by account #10
+		let hash = Kitties::kitty_by_index(0);
+		assert_eq!(Kitties::owner_of(hash), Some(10));
+
+		let other_hash = Kitties::kitty_of_owner_by_index((10, 0));
+		assert_eq!(hash, other_hash);
+	})
+}
+```
+
+Notable things from the above snippet:
+
+- Notice how using a `u64` as `AccountId` is handy. Your account is simply referred to as `10` rather than a special type.
+- Notice how we've used the same `Kitties` to call a getter function (e.g. `kitty_of_owner_by_index`) and a normal dispatch function (e.g. `create_kitty` from `decl_module!`).
+- Notice that you can use `Kitties` to call even the module functions as well (anything inside `impl Module<T>`)! Hence, you could (and should) have a test in which you only test the internal `mint()` function that is not directly accessible.
+
+> **Note:** When accessing storage, you can also use the following alternative syntax in lieu of `Kitties::get()`: 
+
+```rust
+use super::KittyOwner;
+use support::StorageMap;
```

Comment:
Though this depends on the type of storage, so I would make it `use support::{StorageMap, StorageValue, DoubleMap, EnumerableStroageMap}` etc or mention that it __might__ be different.